xml/src/main/java/org/ehcache/xml/ConfigurationParser.java

```diff
@@ -132,6 +156,35 @@ public ConfigurationParser(String xml) throws IOException, SAXException, JAXBExc
     DocumentBuilder domBuilder = factory.newDocumentBuilder();
     domBuilder.setErrorHandler(new FatalErrorHandler());
     Element dom = domBuilder.parse(xml).getDocumentElement();
+
```

Comment:
I would be tempted to go along with @alexsnaps on this one. The scenario described by @chrisdennis even works since for failed replacement, we just leave the original value in place. Only conflict would be if a value in that custom parser has a name clash with a system property. Given that sounds fairly unlikely - let's YAGNI the different handling and decide when the problem pops up.